packages/xrpl/src/sugar/submit.ts

```diff
@@ -177,31 +190,40 @@ async function waitForFinalTransactionOutcome(
 }
 
 // checks if the transaction has been signed
-function isSigned(transaction: Transaction | string): boolean {
-  const tx = typeof transaction === 'string' ? decode(transaction) : transaction
+function isSigned<T extends BaseTransaction = Transaction>(
+  transaction: T | string,
+  definitions?: InstanceType<typeof XrplDefinitionsBase>,
```

Comment:
Should not be optional since this directly uses decode, so we should specify our default one layer above the binary codec across the board.